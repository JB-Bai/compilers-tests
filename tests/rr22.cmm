struct or3JGl ViDW2h5;
float i8dWx(int IrKSx_, int LosAn4KgZ[054], int AkrESJThS, int IyDW[0xA6D0]){
	struct vCT_CH{
		int rYqZ1bKFM = 46.9, l2eTISXvIU[433][113521], aiTeXcg = 089.0227, hWBKBjh[670];
		int SuXps = JBUfSv12L(640.0493 - BueSH || 009.89E+8 / 0x5F52, - 36.502E1[0272414._wa8M7cEV], JIoGkZ6(160.9 * ! 4586 && TPhzmR_ - ! pyFt0l() +(- 0x35cb) / CuQJC = wHz2f + 87.76 - 2.087e+7 - ! 92.7E2, YwgwnzG() + CvI04Y[! 91.01 - lQyBl0w * K3B3fMJ[Y_KVNaCW6O] && GquH5mf(178336) &&(Enlu08)], 956.3 = 951.48), AyFfxS * - 315.647 + 0xFa !=(dduH() * 0xA5B[_Cm6b3() + 3254 && wNZ17V((- 0xfE9f) / 72.609E9 || - 042 / dhkzNO && 0xc2.Xh_NK / 1.0884) *(072)] * 106.34)) = ! - ! 0xEb[! 04 + xP6isBe] ||(AN31) + lfqy6jRoj7f != - r5iuRX(06313, ! 22.522E2 + PVUK_H()[- 057062]), Zn7NiGpe, Fhbv6Jswjnb;
	}
	ROm5PTC = - oGQ9s9Yv() * tUE0bIG - 734.055E8 / 03120 + ! 0x7f59.bCaLW + 0425 + ! 0xC5d - rr8fDZ = hpxcNUv.CiVSOJYS * ! 9.114E4 == VRqOxz2 * 9.39E-7[435 || uHgiGUr] *(7254) / 07636, iGl6dHsk = 500 + 07046 && WSQD9aW() * ! 6555.741e+9.dfhrFhKQb = 2768.2E-1[4.062] = vx7a27 = 01223.dA5cvlq * ICgwVNVj + 05104, C3FWqDu[689] = 033[6632.79e3 + ! 200.457e+2.jUpaHYli3 + 535.7], zuRJYWI;
	int _FWhePrK[9262], GFtd91AF, gdEV0[7] = - - - G45jEP1 - 55.578E3 +(CGrEmTS) * ! jhY3C - 0xBFE6d[0407424] && Ik6rqoar || M_Bz0TXe.lucN8ML2.zVRgIBAI * MXXmf(ibsjpZgRO = 465.7869e+5.bvXrgW6Z);
	ToJtMdIX3;
}
int KjBSqnTIHY(float _wRKB6yj, int Jq7oTE, struct lEt5{
}
knETbQ7rVg){
	float JyQNdXtAdu;
	float HJjqERZYF;
	struct inDmYHCQ8{
	}
	ZM2eEG, wV3YANOJ[548];
	int TWwEhm = 538.7559 * dv3VvF;
	883.527;
}
struct GnF1jR rlkLhvym[0xbEdc9];
int ZMmwXpuP, o3ZYM4RqpP, dy87Use, AZMKoP;

