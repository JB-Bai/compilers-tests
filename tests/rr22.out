Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        StructSpecifier (1)
          STRUCT
          Tag (1)
            ID: or3JGl
      ExtDecList (1)
        VarDec (1)
          ID: ViDW2h5
      SEMI
    ExtDefList (2)
      ExtDef (2)
        Specifier (2)
          TYPE: float
        FunDec (2)
          ID: i8dWx
          LP
          VarList (2)
            ParamDec (2)
              Specifier (2)
                TYPE: int
              VarDec (2)
                ID: IrKSx_
            COMMA
            VarList (2)
              ParamDec (2)
                Specifier (2)
                  TYPE: int
                VarDec (2)
                  VarDec (2)
                    ID: LosAn4KgZ
                  LB
                  INT: 44
                  RB
              COMMA
              VarList (2)
                ParamDec (2)
                  Specifier (2)
                    TYPE: int
                  VarDec (2)
                    ID: AkrESJThS
                COMMA
                VarList (2)
                  ParamDec (2)
                    Specifier (2)
                      TYPE: int
                    VarDec (2)
                      VarDec (2)
                        ID: IyDW
                      LB
                      INT: 42704
                      RB
          RP
        CompSt (2)
          LC
          DefList (3)
            Def (3)
              Specifier (3)
                StructSpecifier (3)
                  STRUCT
                  OptTag (3)
                    ID: vCT_CH
                  LC
                  DefList (4)
                    Def (4)
                      Specifier (4)
                        TYPE: int
                      DecList (4)
                        Dec (4)
                          VarDec (4)
                            ID: rYqZ1bKFM
                          ASSIGNOP
                          Exp (4)
                            FLOAT: 46.900002
                        COMMA
                        DecList (4)
                          Dec (4)
                            VarDec (4)
                              VarDec (4)
                                VarDec (4)
                                  ID: l2eTISXvIU
                                LB
                                INT: 433
                                RB
                              LB
                              INT: 113521
                              RB
                          COMMA
                          DecList (4)
                            Dec (4)
                              VarDec (4)
                                ID: aiTeXcg
                              ASSIGNOP
                              Exp (4)
                                FLOAT: 89.022697
                            COMMA
                            DecList (4)
                              Dec (4)
                                VarDec (4)
                                  VarDec (4)
                                    ID: hWBKBjh
                                  LB
                                  INT: 670
                                  RB
                      SEMI
                    DefList (5)
                      Def (5)
                        Specifier (5)
                          TYPE: int
                        DecList (5)
                          Dec (5)
                            VarDec (5)
                              ID: SuXps
                            ASSIGNOP
                            Exp (5)
                              Exp (5)
                                ID: JBUfSv12L
                                LP
                                Args (5)
                                  Exp (5)
                                    Exp (5)
                                      Exp (5)
                                        FLOAT: 640.049316
                                      MINUS
                                      Exp (5)
                                        ID: BueSH
                                    OR
                                    Exp (5)
                                      Exp (5)
                                        FLOAT: 989000000.000000
                                      DIV
                                      Exp (5)
                                        INT: 24402
                                  COMMA
                                  Args (5)
                                    Exp (5)
                                      MINUS
                                      Exp (5)
                                        Exp (5)
                                          FLOAT: 365.019989
                                        LB
                                        Exp (5)
                                          Exp (5)
                                            INT: 95500
                                          DOT
                                          ID: _wa8M7cEV
                                        RB
                                    COMMA
                                    Args (5)
                                      Exp (5)
                                        ID: JIoGkZ6
                                        LP
                                        Args (5)
                                          Exp (5)
                                            Exp (5)
                                              Exp (5)
                                                Exp (5)
                                                  FLOAT: 160.899994
                                                STAR
                                                Exp (5)
                                                  NOT
                                                  Exp (5)
                                                    INT: 4586
                                              AND
                                              Exp (5)
                                                Exp (5)
                                                  Exp (5)
                                                    ID: TPhzmR_
                                                  MINUS
                                                  Exp (5)
                                                    NOT
                                                    Exp (5)
                                                      ID: pyFt0l
                                                      LP
                                                      RP
                                                PLUS
                                                Exp (5)
                                                  Exp (5)
                                                    LP
                                                    Exp (5)
                                                      MINUS
                                                      Exp (5)
                                                        INT: 13771
                                                    RP
                                                  DIV
                                                  Exp (5)
                                                    ID: CuQJC
                                            ASSIGNOP
                                            Exp (5)
                                              Exp (5)
                                                Exp (5)
                                                  Exp (5)
                                                    ID: wHz2f
                                                  PLUS
                                                  Exp (5)
                                                    FLOAT: 87.760002
                                                MINUS
                                                Exp (5)
                                                  FLOAT: 20870000.000000
                                              MINUS
                                              Exp (5)
                                                NOT
                                                Exp (5)
                                                  FLOAT: 9270.000000
                                          COMMA
                                          Args (5)
                                            Exp (5)
                                              Exp (5)
                                                ID: YwgwnzG
                                                LP
                                                RP
                                              PLUS
                                              Exp (5)
                                                Exp (5)
                                                  ID: CvI04Y
                                                LB
                                                Exp (5)
                                                  Exp (5)
                                                    Exp (5)
                                                      Exp (5)
                                                        NOT
                                                        Exp (5)
                                                          FLOAT: 91.010002
                                                      MINUS
                                                      Exp (5)
                                                        Exp (5)
                                                          ID: lQyBl0w
                                                        STAR
                                                        Exp (5)
                                                          Exp (5)
                                                            ID: K3B3fMJ
                                                          LB
                                                          Exp (5)
                                                            ID: Y_KVNaCW6O
                                                          RB
                                                    AND
                                                    Exp (5)
                                                      ID: GquH5mf
                                                      LP
                                                      Args (5)
                                                        Exp (5)
                                                          INT: 178336
                                                      RP
                                                  AND
                                                  Exp (5)
                                                    LP
                                                    Exp (5)
                                                      ID: Enlu08
                                                    RP
                                                RB
                                            COMMA
                                            Args (5)
                                              Exp (5)
                                                Exp (5)
                                                  FLOAT: 956.299988
                                                ASSIGNOP
                                                Exp (5)
                                                  FLOAT: 951.479980
                                        RP
                                      COMMA
                                      Args (5)
                                        Exp (5)
                                          Exp (5)
                                            Exp (5)
                                              Exp (5)
                                                ID: AyFfxS
                                              STAR
                                              Exp (5)
                                                MINUS
                                                Exp (5)
                                                  FLOAT: 315.647003
                                            PLUS
                                            Exp (5)
                                              INT: 250
                                          RELOP
                                          Exp (5)
                                            LP
                                            Exp (5)
                                              Exp (5)
                                                Exp (5)
                                                  ID: dduH
                                                  LP
                                                  RP
                                                STAR
                                                Exp (5)
                                                  Exp (5)
                                                    INT: 2651
                                                  LB
                                                  Exp (5)
                                                    Exp (5)
                                                      Exp (5)
                                                        ID: _Cm6b3
                                                        LP
                                                        RP
                                                      PLUS
                                                      Exp (5)
                                                        INT: 3254
                                                    AND
                                                    Exp (5)
                                                      Exp (5)
                                                        ID: wNZ17V
                                                        LP
                                                        Args (5)
                                                          Exp (5)
                                                            Exp (5)
                                                              Exp (5)
                                                                LP
                                                                Exp (5)
                                                                  MINUS
                                                                  Exp (5)
                                                                    INT: 65183
                                                                RP
                                                              DIV
                                                              Exp (5)
                                                                FLOAT: 72608997376.000000
                                                            OR
                                                            Exp (5)
                                                              Exp (5)
                                                                Exp (5)
                                                                  MINUS
                                                                  Exp (5)
                                                                    INT: 34
                                                                DIV
                                                                Exp (5)
                                                                  ID: dhkzNO
                                                              AND
                                                              Exp (5)
                                                                Exp (5)
                                                                  Exp (5)
                                                                    INT: 194
                                                                  DOT
                                                                  ID: Xh_NK
                                                                DIV
                                                                Exp (5)
                                                                  FLOAT: 1.088400
                                                        RP
                                                      STAR
                                                      Exp (5)
                                                        LP
                                                        Exp (5)
                                                          INT: 58
                                                        RP
                                                  RB
                                              STAR
                                              Exp (5)
                                                FLOAT: 106.339996
                                            RP
                                RP
                              ASSIGNOP
                              Exp (5)
                                Exp (5)
                                  NOT
                                  Exp (5)
                                    MINUS
                                    Exp (5)
                                      NOT
                                      Exp (5)
                                        Exp (5)
                                          INT: 235
                                        LB
                                        Exp (5)
                                          Exp (5)
                                            NOT
                                            Exp (5)
                                              INT: 4
                                          PLUS
                                          Exp (5)
                                            ID: xP6isBe
                                        RB
                                OR
                                Exp (5)
                                  Exp (5)
                                    Exp (5)
                                      LP
                                      Exp (5)
                                        ID: AN31
                                      RP
                                    PLUS
                                    Exp (5)
                                      ID: lfqy6jRoj7f
                                  RELOP
                                  Exp (5)
                                    MINUS
                                    Exp (5)
                                      ID: r5iuRX
                                      LP
                                      Args (5)
                                        Exp (5)
                                          INT: 3275
                                        COMMA
                                        Args (5)
                                          Exp (5)
                                            Exp (5)
                                              NOT
                                              Exp (5)
                                                FLOAT: 2252.199951
                                            PLUS
                                            Exp (5)
                                              Exp (5)
                                                ID: PVUK_H
                                                LP
                                                RP
                                              LB
                                              Exp (5)
                                                MINUS
                                                Exp (5)
                                                  INT: 24114
                                              RB
                                      RP
                          COMMA
                          DecList (5)
                            Dec (5)
                              VarDec (5)
                                ID: Zn7NiGpe
                            COMMA
                            DecList (5)
                              Dec (5)
                                VarDec (5)
                                  ID: Fhbv6Jswjnb
                        SEMI
                  RC
              DecList (7)
                Dec (7)
                  VarDec (7)
                    ID: ROm5PTC
                  ASSIGNOP
                  Exp (7)
                    Exp (7)
                      Exp (7)
                        Exp (7)
                          Exp (7)
                            Exp (7)
                              Exp (7)
                                Exp (7)
                                  MINUS
                                  Exp (7)
                                    ID: oGQ9s9Yv
                                    LP
                                    RP
                                STAR
                                Exp (7)
                                  ID: tUE0bIG
                              MINUS
                              Exp (7)
                                Exp (7)
                                  FLOAT: 73405497344.000000
                                DIV
                                Exp (7)
                                  INT: 1616
                            PLUS
                            Exp (7)
                              NOT
                              Exp (7)
                                Exp (7)
                                  INT: 32601
                                DOT
                                ID: bCaLW
                          PLUS
                          Exp (7)
                            INT: 277
                        PLUS
                        Exp (7)
                          NOT
                          Exp (7)
                            INT: 3165
                      MINUS
                      Exp (7)
                        ID: rr8fDZ
                    ASSIGNOP
                    Exp (7)
                      Exp (7)
                        Exp (7)
                          Exp (7)
                            ID: hpxcNUv
                          DOT
                          ID: CiVSOJYS
                        STAR
                        Exp (7)
                          NOT
                          Exp (7)
                            FLOAT: 91140.000000
                      RELOP
                      Exp (7)
                        Exp (7)
                          Exp (7)
                            Exp (7)
                              ID: VRqOxz2
                            STAR
                            Exp (7)
                              Exp (7)
                                FLOAT: 0.000001
                              LB
                              Exp (7)
                                Exp (7)
                                  INT: 435
                                OR
                                Exp (7)
                                  ID: uHgiGUr
                              RB
                          STAR
                          Exp (7)
                            LP
                            Exp (7)
                              INT: 7254
                            RP
                        DIV
                        Exp (7)
                          INT: 3998
                COMMA
                DecList (7)
                  Dec (7)
                    VarDec (7)
                      ID: iGl6dHsk
                    ASSIGNOP
                    Exp (7)
                      Exp (7)
                        Exp (7)
                          Exp (7)
                            INT: 500
                          PLUS
                          Exp (7)
                            INT: 3622
                        AND
                        Exp (7)
                          Exp (7)
                            ID: WSQD9aW
                            LP
                            RP
                          STAR
                          Exp (7)
                            NOT
                            Exp (7)
                              Exp (7)
                                FLOAT: 6555741192192.000000
                              DOT
                              ID: dfhrFhKQb
                      ASSIGNOP
                      Exp (7)
                        Exp (7)
                          Exp (7)
                            FLOAT: 276.820007
                          LB
                          Exp (7)
                            FLOAT: 4.062000
                          RB
                        ASSIGNOP
                        Exp (7)
                          Exp (7)
                            ID: vx7a27
                          ASSIGNOP
                          Exp (7)
                            Exp (7)
                              Exp (7)
                                Exp (7)
                                  INT: 659
                                DOT
                                ID: dA5cvlq
                              STAR
                              Exp (7)
                                ID: ICgwVNVj
                            PLUS
                            Exp (7)
                              INT: 2628
                  COMMA
                  DecList (7)
                    Dec (7)
                      VarDec (7)
                        VarDec (7)
                          ID: C3FWqDu
                        LB
                        INT: 689
                        RB
                      ASSIGNOP
                      Exp (7)
                        Exp (7)
                          INT: 27
                        LB
                        Exp (7)
                          Exp (7)
                            Exp (7)
                              FLOAT: 6632790.000000
                            PLUS
                            Exp (7)
                              NOT
                              Exp (7)
                                Exp (7)
                                  FLOAT: 20045.699219
                                DOT
                                ID: jUpaHYli3
                          PLUS
                          Exp (7)
                            FLOAT: 535.700012
                        RB
                    COMMA
                    DecList (7)
                      Dec (7)
                        VarDec (7)
                          ID: zuRJYWI
              SEMI
            DefList (8)
              Def (8)
                Specifier (8)
                  TYPE: int
                DecList (8)
                  Dec (8)
                    VarDec (8)
                      VarDec (8)
                        ID: _FWhePrK
                      LB
                      INT: 9262
                      RB
                  COMMA
                  DecList (8)
                    Dec (8)
                      VarDec (8)
                        ID: GFtd91AF
                    COMMA
                    DecList (8)
                      Dec (8)
                        VarDec (8)
                          VarDec (8)
                            ID: gdEV0
                          LB
                          INT: 7
                          RB
                        ASSIGNOP
                        Exp (8)
                          Exp (8)
                            Exp (8)
                              Exp (8)
                                Exp (8)
                                  Exp (8)
                                    MINUS
                                    Exp (8)
                                      MINUS
                                      Exp (8)
                                        MINUS
                                        Exp (8)
                                          ID: G45jEP1
                                  MINUS
                                  Exp (8)
                                    FLOAT: 55578.000000
                                PLUS
                                Exp (8)
                                  Exp (8)
                                    LP
                                    Exp (8)
                                      ID: CGrEmTS
                                    RP
                                  STAR
                                  Exp (8)
                                    NOT
                                    Exp (8)
                                      ID: jhY3C
                              MINUS
                              Exp (8)
                                Exp (8)
                                  INT: 786029
                                LB
                                Exp (8)
                                  INT: 134932
                                RB
                            AND
                            Exp (8)
                              ID: Ik6rqoar
                          OR
                          Exp (8)
                            Exp (8)
                              Exp (8)
                                Exp (8)
                                  ID: M_Bz0TXe
                                DOT
                                ID: lucN8ML2
                              DOT
                              ID: zVRgIBAI
                            STAR
                            Exp (8)
                              ID: MXXmf
                              LP
                              Args (8)
                                Exp (8)
                                  Exp (8)
                                    ID: ibsjpZgRO
                                  ASSIGNOP
                                  Exp (8)
                                    Exp (8)
                                      FLOAT: 46578688.000000
                                    DOT
                                    ID: bvXrgW6Z
                              RP
                SEMI
          StmtList (9)
            Stmt (9)
              Exp (9)
                ID: ToJtMdIX3
              SEMI
          RC
      ExtDefList (11)
        ExtDef (11)
          Specifier (11)
            TYPE: int
          FunDec (11)
            ID: KjBSqnTIHY
            LP
            VarList (11)
              ParamDec (11)
                Specifier (11)
                  TYPE: float
                VarDec (11)
                  ID: _wRKB6yj
              COMMA
              VarList (11)
                ParamDec (11)
                  Specifier (11)
                    TYPE: int
                  VarDec (11)
                    ID: Jq7oTE
                COMMA
                VarList (11)
                  ParamDec (11)
                    Specifier (11)
                      StructSpecifier (11)
                        STRUCT
                        OptTag (11)
                          ID: lEt5
                        LC
                        RC
                    VarDec (13)
                      ID: knETbQ7rVg
            RP
          CompSt (13)
            LC
            DefList (14)
              Def (14)
                Specifier (14)
                  TYPE: float
                DecList (14)
                  Dec (14)
                    VarDec (14)
                      ID: JyQNdXtAdu
                SEMI
              DefList (15)
                Def (15)
                  Specifier (15)
                    TYPE: float
                  DecList (15)
                    Dec (15)
                      VarDec (15)
                        ID: HJjqERZYF
                  SEMI
                DefList (16)
                  Def (16)
                    Specifier (16)
                      StructSpecifier (16)
                        STRUCT
                        OptTag (16)
                          ID: inDmYHCQ8
                        LC
                        RC
                    DecList (18)
                      Dec (18)
                        VarDec (18)
                          ID: ZM2eEG
                      COMMA
                      DecList (18)
                        Dec (18)
                          VarDec (18)
                            VarDec (18)
                              ID: wV3YANOJ
                            LB
                            INT: 548
                            RB
                    SEMI
                  DefList (19)
                    Def (19)
                      Specifier (19)
                        TYPE: int
                      DecList (19)
                        Dec (19)
                          VarDec (19)
                            ID: TWwEhm
                          ASSIGNOP
                          Exp (19)
                            Exp (19)
                              FLOAT: 538.755920
                            STAR
                            Exp (19)
                              ID: dv3VvF
                      SEMI
            StmtList (20)
              Stmt (20)
                Exp (20)
                  FLOAT: 883.526978
                SEMI
            RC
        ExtDefList (22)
          ExtDef (22)
            Specifier (22)
              StructSpecifier (22)
                STRUCT
                Tag (22)
                  ID: GnF1jR
            ExtDecList (22)
              VarDec (22)
                VarDec (22)
                  ID: rlkLhvym
                LB
                INT: 781769
                RB
            SEMI
          ExtDefList (23)
            ExtDef (23)
              Specifier (23)
                TYPE: int
              ExtDecList (23)
                VarDec (23)
                  ID: ZMmwXpuP
                COMMA
                ExtDecList (23)
                  VarDec (23)
                    ID: o3ZYM4RqpP
                  COMMA
                  ExtDecList (23)
                    VarDec (23)
                      ID: dy87Use
                    COMMA
                    ExtDecList (23)
                      VarDec (23)
                        ID: AZMKoP
              SEMI
